include/VecCore/Backend/Implementation.h

```diff
@@ -284,6 +285,21 @@ VECCORE_FORCE_INLINE VECCORE_ATT_HOST_DEVICE constexpr Bool_s EarlyReturnAllowed
 #endif
 }
 
+// An extension of EarlyReturnAllowed(), and new name is a reminder that it can be used for other purposes as well
+template <typename T, size_t simdSize = 1>
+VECCORE_FORCE_INLINE
+VECCORE_ATT_HOST_DEVICE
+constexpr Bool_s SIMDsizeUpTo()
```

Comment:
> This function looks a bit backwards also, since it always returns false on CUDA. For instance, SIMDsizeUpTo<double, 1>() should return true everywhere. What we want is to check that the length is wide enough to get gains in vector code.

Yes and on CUDA, at the moment the vector size is always one *and* we are claiming that making the thread going out-of-sync (which any early return would do) is more costly than continuing ...

So you are right that the routines implementation is exactly
```
return vecCore::EarlyReturnAllowed() && vecCore::VectorSize<T>() <= simdSize;
```
I.e. the intent is 'if allowed, let me early return the number of element in the vector is smaller than the number I provide (i.e. usually 1 to allow only scalar but could be different if the author 'known' that the likelihood of maskFull(vec) being true is high enough (for a the choosen vector size) to be worth the calculation.